share/lib/python/neuron/rxd/rxd.py

```diff
@@ -1250,17 +1250,26 @@ def localize_index(creg, rate):
                     species_id = creg._species_ids[s._id]
                     for reg in creg._react_regions[rptr]:
                         if reg() in r._rate:
-                            region_id = creg._region_ids[reg()._id]
-                            rate_str = localize_index(creg, r._rate[reg()][0])
+                            try:
+                                region_id = creg._region_ids[reg()._id]
+                                rate_str = localize_index(creg, r._rate[reg()][0])
+                            except KeyError:
+                                warn("Species not on the region specified, %r will be ignored.\n" % r)
+                                continue
                             operator = '+=' if species_ids_used[species_id][region_id] else '='
                             fxn_string += "\n\trhs[%d][%d] %s %s;" % (species_id, region_id, operator, rate_str)
                             species_ids_used[species_id][region_id] = True
                 elif isinstance(r, multiCompartmentReaction.MultiCompartmentReaction):
                     #Lookup the region_id for the reaction
-                    for reg in r._rate:
-                        rate_str = localize_index(creg, r._rate[reg][0])
-                        fxn_string += "\n\trate = %s;" % rate_str
-                        break
+                    try:
+                        for reg in r._rate:
+                            rate_str = localize_index(creg, r._rate[reg][0])
+                            fxn_string += "\n\trate = %s;" % rate_str
+                            break
+                    except KeyError:
+                        warn("Species not on the region specified, %r will be ignored.\n" % r)
+                        continue
```

Comment:
Does this work? Shouldn't the try/except be inside the for loop?

Reply:
My mistake. Continue here moves to the next reaction, which is exactly the right thing to do.